### PR TITLE
Decompile t-rex, little spiders, big spiders, yeti, and bird guardian

### PR DESCRIPTION
**Pull request by rr-**
#### Checklist

- [X] I have read the [coding conventions](https://github.com/LostArtefacts/TR1X/blob/master/CONTRIBUTING.md#coding-conventions)
- [X] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

#1693
This should cover all non-aquatic and non-human enemies.

### CODE REVIEW

**Review by aredfan — APPROVED** [review]

File: src/tr2/game/objects/creatures/trex.c [hunk 1]
```diff
@@ -32,3 +65,104 @@ void TRex_Setup(void)
...
+        Creature_Mood(item, &info, MOOD_ATTACK);
+        angle = Creature_Turn(item, creature->maximum_turn);
+        if (item->touch_bits != 0) {
+            g_LaraItem->hit_points -= item->current_anim_state == TREX_STATE_RUN
+                ? TREX_TRAMPLE_DAMAGE
+                : TREX_TOUCH_DAMAGE;
+        }
+
+        creature->flags = creature->mood != MOOD_ESCAPE && !info.ahead
+            && info.enemy_facing > TREX_FRONT_ARC
```
Comment (lahm86) [inline src/tr2/game/objects/creatures/trex.c]: I'm wondering if this should be `-TREX_FRONT_ARC` - currently the t-rex tends to just circle Lara.

File: src/tr2/game/objects/creatures/yeti.c [hunk 2]
```diff
@@ -32,3 +93,233 @@ void Yeti_Setup(void)
...
+                item->goal_anim_state = YETI_STATE_STOP;
+            }
+            break;
+
+        case YETI_STATE_WAIT_2:
+            if (creature->mood == MOOD_ESCAPE || item->hit_status) {
+                item->goal_anim_state = YETI_STATE_STOP;
+            } else if (creature->mood == MOOD_BORED) {
+                const int32_t random = Random_GetControl();
+                if (random < YETI_WAIT_1_CHANCE && lara_alive) {
```
Comment (lahm86) [inline src/tr2/game/objects/creatures/yeti.c]: Just a sanity check as this is `||  !lara_alive` in `YETI_STATE_STOP` above - I can't see any behaviour differences but just thought it worth to double check.

File: src/tr2/game/objects/creatures/yeti.c [hunk 3]
```diff
@@ -32,3 +93,233 @@ void Yeti_Setup(void)
...
+            }
+            break;
+
+        case YETI_STATE_WALK:
+            creature->maximum_turn = YETI_WALK_TURN;
+            if (creature->mood == MOOD_ESCAPE) {
+                item->goal_anim_state = YETI_STATE_RUN;
+            } else if (creature->mood == MOOD_BORED) {
+                const int32_t random = Random_GetControl();
+                if (random < YETI_WAIT_1_CHANCE && lara_alive) {
```
Comment (lahm86) [inline src/tr2/game/objects/creatures/yeti.c]: And similar here.

File: src/tr2/game/objects/creatures/trex.c (hunk 1 above)
Comment (rr-) [reply src/tr2/game/objects/creatures/trex.c]: Great catch! That was a regression I introduced. Thank you.

File: src/tr2/game/objects/creatures/yeti.c (hunk 2 above)
Comment (rr-) [reply src/tr2/game/objects/creatures/yeti.c]: Same as above.

File: src/tr2/game/objects/creatures/yeti.c (hunk 3 above)
Comment (rr-) [reply src/tr2/game/objects/creatures/yeti.c]: Same as above.

**Review by lahm86 — APPROVED** [review]

**Pull request merged**